design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+
+---
+Wireframes
+
+[Item Type Warning](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.22&scaling=min-zoom&starting-point-node-id=8268%3A90606&show-proto-sidebar=1)
+
+[All Column Item Change Warning](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.4&scaling=min-zoom&starting-point-node-id=8268%3A90982&show-proto-sidebar=1)
+
+---
+
+## 2. Converting Existing List Data to a Different Type
+
+Users can perform numerous data type operations to List data. There are two main categories of conversion:
+
+1. Converting List data to a basic data type as a whole.
+2. Converting List items to a different basic data type.
+
+### 2.1. Converting List Data to a Basic Type by Turning Off `Allow List`
+
+By default, the `Allow List` toggle in the column header menu would be turned on. While users are editing data cells, if Mathesar detects a comma delimiter and a data structure that resembles a List, Mathesar can make automatically inference and convert the column into a List data type. Users can still maintain the same basic data type for each List item, and the common basic data type settings, such as formatting and setting up Null values, are still available for List items.
+
+Users can convert the entire List back to a basic type by simply turning off the `Allow List` toggle. While the toggle is turned off, the column will not automatically and cannot be converted to a List data type. In this case, users' comma delimiter formatted data can only be recognized as Text by Mathesar.
+
+---
+Wireframes
+
+[Turning Off `Allow List`](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.34&scaling=min-zoom&starting-point-node-id=8268%3A91375&show-proto-sidebar=1)
+
+---
+
+### 2.2 Converting List Items to a Different Type
+
+Users can keep the List data structure and convert the List item data type only. As with all data entries, all List items can be converted into Text items. Mathesar will check whether the item format comforms to the destination data type requirement for all other conversions. Users will be warned about potential data loss as required. The conversion between different data types has been explored and will be detailed in a separate spec. The wireframe example is demonstrating changing List items from Number to Text.
+
+---
+Wireframes
+
+[Converting List Items](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.26&scaling=min-zoom&starting-point-node-id=8268%3A91011&show-proto-sidebar=1)
+
+---
+
+## 3. Creating a New Column in List Data Type
+
+Since Mathesar aims to support all basic data types in List format as well as detect List data structure automatically, List data capability comes with every new column added. Users can opt out this feature by turning off `Allow List`.
+
```

Comment:
I believe that in the majority of cases, users will not want lists for all columns because it will make it difficult to identify relationships between data items. I think list will be an exemption for a few cases, so I'm not sure we want to enable the option by default.

Reply:
This is a great point. Turning on List option does not convert data to List if List structure is not detected. However, Kriti mentioned Mathesar didn't have the capability to check data in realtime – I need to reconsider this design.